pandas/tests/dtypes/cast/test_infer_dtype.py

```diff
@@ -187,14 +188,29 @@ def test_infer_dtype_from_array(arr, expected, pandas_dtype):
         (1.1, np.float64),
         (Timestamp("2011-01-01"), "datetime64[ns]"),
         (Timestamp("2011-01-01", tz="US/Eastern"), object),
-        (Period("2011-01-01", freq="D"), object),
     ],
 )
-def test_cast_scalar_to_array(obj, dtype):
+def test_cast_scalar_to_numpy_array(obj, dtype):
     shape = (3, 2)
 
     exp = np.empty(shape, dtype=dtype)
     exp.fill(obj)
 
     arr = cast_scalar_to_array(shape, obj, dtype=dtype)
     tm.assert_numpy_array_equal(arr, exp)
+
+
+@pytest.mark.parametrize(
+    "obj,dtype",
+    [
+        (Period("2011-01-01", freq="D"), PeriodDtype('D')),
+        (Period("2011-01", freq="M"), PeriodDtype('M')),
```

Comment:
I don't think we need multiple tests for Periods here. It would be useful to have tests for other extension dtypes.

1. `Interval`.
2. `Timestamp` (with and without a timezone)

Reply:
Can you also add "user-facing" tests in `pandas/tests/frame/indexing/test_setitem.py` that do something like the example from the issue?

Reply:
I added an Interval value to the existing test and added a user-facing test in `test_setitem`.

I actually avoided making my `test_cast_scalar_to_extension_array` include Timestamp variables since those are actually covered currently in the `test_cast_scalar_to_numpy_array` test that is above mine. I think the logic here is that timestamps can be converted to numpy timestamp and thus don't need to use an extension array. Please let me know if you would prefer this another way.

Reply:
> timestamps can be converted to numpy timestamp and thus don't need to use an extension array

timezone-naive timestamps can be converted to numpy. But Timestamps with a timezone can't be, and so we have an extension array for tz-aware timestamps.

Reply:
Got it, added that to a Timestamp with a timezone to the test!